### ISSUE TITLE
Understanding docker and port access trough the internet

### ISSUE THREAD

**Issue by phlegx** [docker/docker#6507]
Starting a container (docker registry server) like in this <b>first case</b>:

```bash
docker run -d -p 5000:5000 samalba/docker-registry
```

allows me to access the docker registry from every other host in the web.

Starting a container (docker registry server) like in this <b>second case</b>:

```bash
docker run -d -p 127.0.0.1:5000:5000 samalba/docker-registry
```

Then NO more access is possible from every other host in the web.

I was wondering now what is the best option: listen to localhost only or listen to alll interfaces? And second, regarding the <b>first case</b> when I listen to all interfaces: How can I restrict the access through iptables and allow only certain hosts to access this port 5000 from outside the (host) machine? I tried stuff like for example:

```bash
-A FORWARD -s 95.xx.xx.xx/32 -p tcp -m tcp --dport 5000 -j ACCEPT
-A FORWARD -p tcp --destination-port 5000 -j DROP
```
with unfortunately no result. I still can access port 5000 from all over the internet. 

I also tried to limit it with an INPUT rule like:

```bash
-A INPUT -s 95.xx.xx.xx/32 -p tcp -m tcp --dport 5000 -j ACCEPT
-A INPUT -j DROP
```

but also this has no effect. 5000 is still accessible for everyone. 

can someone please  tell me what I'm doing wrong?

**Comment by jfrazelle** [comment]
@gdm85 was this resolved by #6034 or no?

**Comment by jfrazelle** [comment]
closing as stale please ping me with exact steps to reproduce on latest and I will reopen
